scripts/generator/graphql_generator/csharp/SDK/DesktopMessageReceiver.cs.erb

```diff
@@ -0,0 +1,72 @@
+namespace <%= namespace %>.SDK {
+    using System;
+    using System.Runtime.InteropServices;
+    using Shopify.Unity.SDK;
+    using Shopify.Unity.GraphQL;
+
+    #if !SHOPIFY_MONO_UNIT_TEST
+    using UnityEngine;
+    public partial class DesktopMessageReceiver: MonoBehaviour {}
+    #endif
+
+    public partial class DesktopMessageReceiver {
+        public ShopifyClient Client;
+        public Checkout CurrentCheckout;
+
+        public CheckoutSuccessCallback OnSuccess;
+        public CheckoutCancelCallback OnCancelled;
+        public CheckoutFailureCallback OnFailure;
+
+        private bool isPaused = false;
+
+        public void Init(
+            ShopifyClient client,
+            Checkout currentCheckout,
+            CheckoutSuccessCallback onSuccess,
+            CheckoutCancelCallback onCancelled,
+            CheckoutFailureCallback onFailure)
+        {
+            Debug.Log("DesktopMessageReceiver Init");
+
+            Client = client;
+            CurrentCheckout = currentCheckout;
+            OnSuccess = onSuccess;
+            OnCancelled = onCancelled;
+            OnFailure = onFailure;
+        }
+
+        void OnApplicationFocus(bool hasFocus)
+        {
+            isPaused = !hasFocus;
+
+            Debug.Log("DesktopMessageReceiver OnApplicationFocus");
+
+            if (!isPaused) {
+                var query = new QueryRootQuery();
+                query.node(
+                    buildQuery: n => n
+                    .onCheckout(c => c.completedAt()),
+                    id: CurrentCheckout.id()
+                );
+
+                Client.Query(query, (response, error) => {
+                    if (error != null) {
+                        OnFailure(error);
+                    } else {
+                        var checkout = (Checkout) response.node();
+                        if (checkout.completedAt() != null) {
+                            OnSuccess();
+                        } else {
+                            OnCancelled();
+                        }
+                    }
+                });
+            }
+        }
+
+        void OnApplicationPause(bool pauseStatus)
```

Comment:
I believe it's it's relevant when on Android, the use case here is for desktop, but since the Unity documentation seems to recommend its use I included it https://docs.unity3d.com/ScriptReference/MonoBehaviour.OnApplicationFocus.html